apps/online_oidc_provider/authentication.py

```diff
@@ -20,3 +23,56 @@ def authenticate(self, request):
             raise exceptions.AuthenticationFailed("The oauth2 token has expired")
 
         return oauth2_token.user, None
+
+
+class TokenHasScope(BasePermission):
```

Comment:
Is there a reason we are writing our own version of `TokenHasScope`, instead of just using DRF's built in one? If so that should be clearly stated in a comment here

Reply:
AFAIK DRF does not have a built-in `TokenHasScope`. We previously used oauth2_provider's `TokenHasScope`, made by Django OAuth Toolkit to provide a support layer for DRF. As we are no longer using oauth2_provider, we cannot use their `TokenHasScope` either.

Reply:
I'm also a bit vary of this. Could you do some more research with regards to this? Could we depend on oauth2_provider for that function alone, or is it tightly coupled to its own OAuth2 provider? Do we want to depend on it, or do we want to run our own fork of the code? At least get some tests up if we do :)

Reply:
Hm, I've been digging through the code for the OAuth2 provider, and the answer is a big _I don't know_. I guess I can continue testing some more, but it seems that if we are to use the TokenHasScope from oauth2_provider, we will need duplicates of settings, to ensure that the oidc_provider and oauth2_provider has the same settings, as TokenHasScope relies on settings for oauth2_provider.

Reply:
The class is by the way taken from this stale issue at oidc_provider's repo https://github.com/juanifioren/django-oidc-provider/issues/342

Other issues related to this worth a read:
https://github.com/juanifioren/django-oidc-provider/issues/242
https://github.com/juanifioren/django-oidc-provider/issues/78